integrations/optimum/src/haystack_integrations/components/embedders/pooling.py

```diff
@@ -0,0 +1,231 @@
+import json
+from enum import Enum
+from typing import Optional
+
+import torch
+from haystack.utils import Secret
+from huggingface_hub import hf_hub_download
+
+
+class PoolingMode(Enum):
+    """
+    Pooling Modes support by the Optimum Embedders.
+    """
+
+    CLS = "cls"
+    MEAN = "mean"
+    MAX = "max"
+    MEAN_SQRT_LEN = "mean_sqrt_len"
+    WEIGHTED_MEAN = "weighted_mean"
+    LAST_TOKEN = "last_token"
+
+    def __str__(self):
+        return self.value
+
+    @classmethod
+    def from_str(cls, string: str) -> "PoolingMode":
+        """
+        Create a pooling mode from a string.
+
+        :param string:
+            The string to convert.
+        :returns:
+            The pooling mode.
+        """
+        enum_map = {e.value: e for e in PoolingMode}
+        pooling_mode = enum_map.get(string)
+        if pooling_mode is None:
+            msg = f"Unknown Pooling mode '{string}'. Supported modes are: {list(enum_map.keys())}"
+            raise ValueError(msg)
+        return pooling_mode
+
+
+class HFPoolingMode:
+    """
+    Gets the pooling mode of the Sentence Transformer model from the Hugging Face Hub.
+    """
+
+    @staticmethod
+    def get_pooling_mode(model: str, token: Optional[Secret] = None) -> Optional[PoolingMode]:
+        try:
+            pooling_config_path = hf_hub_download(repo_id=model, token=token, filename="1_Pooling/config.json")
+
+            with open(pooling_config_path) as f:
+                pooling_config = json.load(f)
+
+            # Filter only those keys that start with "pooling_mode" and are True
+            true_pooling_modes = [
+                key for key, value in pooling_config.items() if key.startswith("pooling_mode") and value
+            ]
+
+            pooling_modes_map = {
+                "pooling_mode_cls_token": PoolingMode.CLS,
+                "pooling_mode_mean_tokens": PoolingMode.MEAN,
+                "pooling_mode_max_tokens": PoolingMode.MAX,
+                "pooling_mode_mean_sqrt_len_tokens": PoolingMode.MEAN_SQRT_LEN,
+                "pooling_mode_weightedmean_tokens": PoolingMode.WEIGHTED_MEAN,
+                "pooling_mode_last_token": PoolingMode.LAST_TOKEN,
+            }
+
+            # If exactly one True pooling mode is found, return it
+            if len(true_pooling_modes) == 1:
+                pooling_mode_from_config = true_pooling_modes[0]
+                pooling_mode = pooling_modes_map.get(pooling_mode_from_config)
+            # If no True pooling modes or more than one True pooling mode is found, return None
+            else:
+                pooling_mode = None
+            return pooling_mode
+        except Exception:
+            return None
+
+
+class Pooling:
```

Comment:
The code in this class is essentially a duplication of the [`Pooling` class in `sentence-transformers`](https://github.com/UKPLab/sentence-transformers/blob/ecdda29859d6a651326addd11bd237870b39866a/sentence_transformers/models/Pooling.py#L100). As such, I see no need for it, especially since the latter is a dependency for this integration.

Let's just reuse their code instead. We can still have the enum and a wrapper class to convert the inputs/outputs to the right formats.

Reply:
Makes sense, now that we are keeping `sentence-transformers` as a dependency.

The individual pooling methods have been removed. The `pool_embeddings` method now converts the inputs and outputs to the right format.